### PR TITLE
Log4J upgrade 2021-02-02 CI

### PR DESCRIPTION
**Pull request by dantewang**

### CODE REVIEW

**Comment by dantewang** [comment]
ci:test:sf

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
ci:test:sf

**Comment by dantewang** [comment]
@zhangyan0701 This pull is going to have lots of failed tests but I think they can be grouped into several areas. Could you scan the result and group the errors by their exception, etc. with the help of Testray or Spira?

**Comment by dantewang** [comment]
The ci-test-sf failures don't seem to be related. Neither of the classes is modified in this pull, and the changes in these classes triggering SF were made by others. For example, the changes in Release.java are done by BChan.

**Comment by dantewang** [comment]
Pushed new commits, which should fix "unit-jdk8", "modules-compile-jdk8" and "modules-semantic-version-jdk8". Looking into "modules-unit-jdk8" now.

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
In this run, the sem ver failure doesn't seem to be related, should be upstream; the modules-compile failure is caused by the Log4J test and Hai is working on it.

**Comment by dantewang** [comment]
I can reproduce the same `NoClassDefFoundError` in modules-unit-jdk8 on master. This should be caused by LPS-126920.

**Comment by GinsonRen** [comment]
@tinatian @dantewang Here is the diff of startup:

Portal Version | Git Id | warmup(ms) | No.1 | No.2 | No.3 | No.4 | No.5 | Average Startup (ms) | Change
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
BaseLine | ee93bfa38 | 45523 | 22,596 | 22,455 | 23,012 | 22,532 | 22,547 | 22,628.4 | -
PR-1901 | - | 46647 | 22875 | 22,714 | 23257 | 22,541 | 22833 | 22844 | 0.95%

**Comment by zhangyan0701** [comment]
Based on the Functional test results on Testray, there are several issues:

**1. Most of the tests failed when delete Virtual Instance by Headless API:**
Example Error 1: https://testray.liferay.com/home/-/testray/case_results/index?delta=200&p_p_state=normal&orderByCol=status_sortable&orderByType=asc&testrayBuildId=1462877199&errors=items
Example Error 2: https://testray.liferay.com/home/-/testray/case_results/index?delta=200&p_p_state=normal&orderByCol=status_sortable&orderByType=asc&testrayBuildId=1462877199&errors=process
Below error in log:
`2021-02-02 10:44:53.073 ERROR [LCS Worker 4][LCSClusterEntryTokenCheckTask:85] Unable to validate the environment token file
     [exec] com.liferay.lcs.client.exception.MissingLCSClusterEntryTokenException: The environment token file is missing. Please download a token file from LCS and place it in /opt/dev/projects/github/liferay-portal/bundles/data
     [exec] 	at com.liferay.lcs.client.internal.advisor.LCSClusterEntryTokenAdvisorImpl.getLCSClusterEntryTokenFileName(LCSClusterEntryTokenAdvisorImpl.java:237) ~[bundleFile:?]
     [exec] 	at com.liferay.lcs.client.internal.advisor.LCSClusterEntryTokenAdvisorImpl._processLCSClusterEntryTokenFile(LCSClusterEntryTokenAdvisorImpl.java:329) ~[bundleFile:?]
     [exec] 	at com.liferay.lcs.client.internal.advisor.LCSClusterEntryTokenAdvisorImpl.processLCSClusterEntryToken(LCSClusterEntryTokenAdvisorImpl.java:121) ~[bundleFile:?]
     [exec] 	at com.liferay.lcs.client.internal.task.LCSClusterEntryTokenCheckTask._checkLCSClusterEntryToken(LCSClusterEntryTokenCheckTask.java:114) ~[bundleFile:?]
     [exec] 	at com.liferay.lcs.client.internal.task.LCSClusterEntryTokenCheckTask.doRun(LCSClusterEntryTokenCheckTask.java:60) [bundleFile:?]
     [exec] 	at com.liferay.lcs.client.internal.task.BaseTask.run(BaseTask.java:35) [bundleFile:?]
     [exec] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_121]
     [exec] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
     [exec] 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_121]
     [exec] 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_121]
     [exec] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_121]
     [exec] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_121]
     [exec] 	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
     [exec]     Timeout occurred while executing commands: [curl -X GET -u 'test@liferay.com:test' -H 'Content-Type: application/json' http://cloud-10-0-10-71.lax.liferay.com:8080/o/headless-portal-instances/v1.0/portal-instances]
     [exec]     BUILD FAILED: process hasn't exited`
**2. Some tests passed themselves, but assert log failed:**
https://testray.liferay.com/home/-/testray/case_results/index?delta=200&p_p_state=normal&orderByCol=status_sortable&orderByType=asc&testrayBuildId=1462877199&errors=thrown
`     [exec]     LIFERAY_ERROR: org.dom4j.DocumentException: Error on line 8 of document file:/opt/dev/projects/github/liferay-portal/bundles/osgi/log4j/com.liferay.portal.osgi.web.wab.generator.impl-log4j-ext.xml : The element type "Loggers" must be terminated by the matching end-tag "</Loggers>".
     [exec]     LIFERAY_ERROR: org.dom4j.DocumentException: Error on line 8 of document file:/opt/dev/projects/github/liferay-portal/bundles/osgi/log4j/com.liferay.portal.search.elasticsearch7.impl-log4j-ext.xml : The element type "Loggers" must be terminated by the matching end-tag "</Loggers>".
     [exec]     LIFERAY_ERROR: org.dom4j.DocumentException: Error on line 8 of document file:/opt/dev/projects/github/liferay-portal/bundles/osgi/log4j/org.apache.aries.jax.rs.whiteboard-log4j-ext.xml : The element type "Loggers" must be terminated by the matching end-tag "</Loggers>".
     [exec]     LIFERAY_ERROR: org.dom4j.DocumentException: Error on line 6 of document file:/opt/dev/projects/github/liferay-portal/bundles/osgi/log4j/com.liferay.portal.search-log4j-ext.xml : The element type "Loggers" must be terminated by the matching end-tag "</Loggers>".`
3. Some problems in Workflow:

- Metrics page keeps loading: https://testray.liferay.com/reports/production/logs/test-1-15/1612259187027/test-portal-acceptance-pullrequest(master)/3833/functional-tomcat90-mysql57-jdk8/5/0/LocalFile.WorkflowMetrics_ViewWorkloadByAssigneePage/index.html.gz
- Completed task is not visible in My Workflow Tasks: https://testray.liferay.com/reports/production/logs/test-1-18/1612243016319/test-portal-acceptance-pullrequest(master)/5182/functional-upgrade-tomcat90-mysql57-jdk8/0/7/LocalFile.WorkflowUpgrade_ViewWorkflowArchive625/index.html.gz

4. Issue in App Builder, Continue button doesn't work when add Custom Object: https://testray.liferay.com/reports/production/logs/test-1-15/1612259187027/test-portal-acceptance-pullrequest(master)/3833/functional-tomcat90-mysql57-jdk8/3/12/LocalFile.SharedAppBuilder_DeleteFormViewFieldFromObject/index.html.gz
5. Some times add portlet to page load very long time: https://testray.liferay.com/reports/production/logs/test-1-18/1612243016319/test-portal-acceptance-pullrequest(master)/5182/functional-tomcat90-mysql57-jdk8/2/23/LocalFile.LocalizationWithWebContentUI_AddLocalizedWebContentViaSourceEditor/index.html.gz
6. Changing locale doesn't work, Japanese content displays even when the locale is en: https://testray.liferay.com/reports/production/logs/test-1-15/1612259187027/test-portal-acceptance-pullrequest(master)/3833/functional-tomcat90-mysql57-jdk8/2/23/LocalFile.LocalizationWithWebContentUI_AddLocalizedWebContentViaSourceEditor/index.html.gz

**Comment by dantewang** [comment]
@GinsonRen @zhangyan0701 Thanks!

**Comment by dantewang** [comment]
Continue in https://github.com/dantewang/liferay-portal/pull/1902
